### PR TITLE
GH-107518: Remove the Argument Clinic How-To

### PR DESCRIPTION
**Pull request by AA-Turner**
* Issue: gh-107518

----
:books: Documentation preview :books:: https://cpython-previews--109900.org.readthedocs.build/

### CODE REVIEW

**Comment by AlexWaygood** [comment]
I'm pretty low on bandwidth right now and haven't been keeping up with this effort; I'll leave this one in @erlend-aasland's highly capable hands, if that's okay :)

File: Doc/howto/clinic.rst
Comment (erlend-aasland) [inline Doc/howto/clinic.rst:1]: Is it possible to install a devguide redirect for this URL?

Comment (AA-Turner) [reply Doc/howto/clinic.rst:1]: We haven't done so for other redirections (e.g. installing --> PPUG), though we could look into it. For the PEPs, it required setting up a custom Terraform configuration for the server, so we'd need to ask someone who understood how the infrastructure is linked up here.

A

Comment (erlend-aasland) [reply Doc/howto/clinic.rst:1]: Ok. I'm not sure it matters much for this particular migration, though.

**Review by erlend-aasland — APPROVED** [review]

**Review by pradyunsg — APPROVED** [review]

**Review by ezio-melotti — APPROVED** [review]

**Pull request merged**